src/service/viewer-impl.js

```diff
@@ -67,19 +67,27 @@ export const ViewportType = {
 /**
  * Visibility state of the AMP document.
  * @enum {string}
- * @private
  */
 export const VisibilityState = {
-
   /**
-   * Viewer has shown the AMP document.
+   * The AMP document is currently active and visible.
    */
   VISIBLE: 'visible',
 
   /**
-   * Viewer has indicated that AMP document is hidden.
+   * The AMP document is active but the browser tab is not.
+   */
+  HIDDEN: 'hidden',
```

Comment:
Do we need also a `PRERENDER` state? Visibility API has it and we definitely have a notion of it in some places in the code - I think it'd be a good idea. Obviously, viewer is not ready for this yet, but here's what we can do in the meantime (based on what you already did below):

```
viewer.visibilityState = hidden:
- hasEverBeenVisible -> PAUSED
- !hasEverBeenVisible -> PRERENDER
```

With doc state the mapping is essentially one-to-one since they have "prerender" state.

I think I'm still a bit confused about UNLOADED. Could you please clarify it a bit more? Specifically, the different between PAUSED and UNLOADED.